common/flatpak-installation.c

```diff
@@ -991,6 +991,75 @@ _ostree_collection_ref_free0 (OstreeCollectionRef *ref)
   ostree_collection_ref_free (ref);
 }
 
+static gboolean
+flatpak_installation_check_installed_ref_missing_related_ref (FlatpakInstallation *self,
+                                                              const gchar         *remote_name,
+                                                              gchar               *full_ref,
+                                                              GCancellable        *cancellable)
+{
+  g_autoptr (GPtrArray) installed_related_refs = NULL;
+  g_autoptr (GPtrArray) remote_related_refs = NULL;
+  g_autoptr (GError) local_error = NULL;
+  guint j, k;
+
+  installed_related_refs = flatpak_installation_list_installed_related_refs_sync (self,
+                                                                                  remote_name,
+                                                                                  full_ref,
+                                                                                  cancellable,
+                                                                                  &local_error);
+  if (local_error != NULL)
+    {
+      g_warning ("Unable to get installed related refs for %s: %s", full_ref, local_error->message);
+      return FALSE;
+    }
+
+  remote_related_refs = flatpak_installation_list_remote_related_refs_sync (self,
+                                                                            remote_name,
+                                                                            full_ref,
+                                                                            cancellable,
+                                                                            &local_error);
+  if (local_error != NULL)
+    {
+      g_warning ("Unable to get remote related refs for %s: %s", full_ref, local_error->message);
+      return FALSE;
+    }
+
+  if (remote_related_refs == NULL)
+    return FALSE;
+
+  for (j = 0; j < remote_related_refs->len; j++)
+    {
+      FlatpakRelatedRef *remote_related_ref = g_ptr_array_index (remote_related_refs, j);
+      g_autofree gchar *remote_related_full_ref = flatpak_ref_format_ref (FLATPAK_REF (remote_related_ref));
+      gboolean should_download = flatpak_related_ref_should_download (remote_related_ref);
+      gboolean related_ref_found = FALSE;
+
+      if (should_download && installed_related_refs != NULL)
```

Comment:
I don't think `installed_related_refs` can be `NULL` here.

Reply:
hmm, it might be look like the above `if (local_error != NULL)` is a good enough safety net but I am panaroid about `installed_related_refs` being passed on as `NULL` without the error being set somewhere down the line. Hence, I introduced ` != NULL` check to avoid a  NULL-de-reference at all cost.

Reply:
Put a g_return_if_fail after the call + check above, then, to make clear that it cannot legitimately be null without error set. Including the check here makes it harder to reason about what the set of possible conditions is.